qa/tasks/cephfs/test_snap_schedules.py

```diff
@@ -577,6 +579,400 @@ def test_snap_schedule_all_periods(self):
         self.mount_a.run_shell(['rmdir', test_dir])
 
 
+class TestSnapSchedulesSubvolAndGroupArguments(TestSnapSchedulesHelper):
+    def setUp(self):
+        super(TestSnapSchedulesSubvolAndGroupArguments, self).setUp()
+        self.CREATE_VERSION = (random.randrange(10) % 2) + 1
```

Comment:
Hmmm.. maybe choose the version via a yaml? That would give deterministic behaviour when rerunning failed tests.

Reply:
AFAICT, yaml fragments do not get past `cephfs_test_runner.py`. The actual python test doesn't get access to any yaml context.

Reply:
Not the yaml fragments themselves, but the config they set should be available via `self.client_config.get(...)`.

C.f.: v1 vs v2 kernel mount syntax - KernelMount::syntax_style.

Reply:
Or maybe I could derive classes ...v1 and ...v2 out of the current implementation and set CREATE_VERSION to 1 and 2 respectively and then run those test cases for the chosen version.
And maybe I could additionally fork off the above tests into two separate files to test v1 and v2 subvolumes with snap-schedule.
e.g. `test_snap_schedules_with_subvolume_v1.py` and `test_snap_schedules_with_subvolume_v2.py`
With this strategy we could test snapshot creation as well within the required versioned subvolume.

Reply:
This is all good object-oriented development, however I'm not sure whether the tests from the base class will be picked up for execution when the `setUp()/tearDown()` methods are removed from the base class and just the CREATE_VERSION = 1 or 2 is the only line found in the derived class `setUp()` method.

Reply:
> Not the yaml fragments themselves, but the config they set should be available via `self.client_config.get(...)`.
> 
> C.f.: v1 vs v2 kernel mount syntax - KernelMount::syntax_style.

client_config is associated with file system client i.e. a mount, like so ...
`self.mount_a.client_config.*`

Reply:
looks like client_config has got nothing to do with test configuration

Reply:
So, `config` is available as part of a teuthology task (e.g. qa/tasks/kclient.py). `cephfs_test_runner` is a teuthology task too which internally runs python tests. Wouldn't it suffice to place a yaml fragment with either v1 or v2 subvolume for the test runner?

Reply:
So there is a `mount_a.ctx` avaialble as well. I'm peering into it to see how I can plug in the yaml to specify the subvol version to create.

Reply:
@mchangir I presume the latest revision picks the version from the yaml fragment?